stl/inc/xutility

```diff
@@ -3444,20 +3514,29 @@ public:
     static constexpr bool _Unwrap_when_unverified = _Do_unwrap_when_unverified_v<iterator_type>;
 
     template <class _Src, enable_if_t<_Wrapped_seekable_v<iterator_type, const _Src&>, int> = 0>
-    constexpr void _Seek_to(const move_iterator<_Src>& _It) {
+    constexpr void _Seek_to(const move_iterator<_Src>& _It) noexcept(noexcept(_Current._Seek_to(_It._Get_current()))) {
         _Current._Seek_to(_It.base());
     }
     template <class _Src, enable_if_t<_Wrapped_seekable_v<iterator_type, _Src>, int> = 0>
-    constexpr void _Seek_to(move_iterator<_Src>&& _It) {
-        _Current._Seek_to(_STD move(_It).base());
+    constexpr void _Seek_to(move_iterator<_Src>&& _It) noexcept(
+        noexcept(_Current._Seek_to(_STD move(_It)._Get_current()))) {
+        _Current._Seek_to(_STD move(_It)._Get_current());
+    }
+
+    constexpr const iterator_type& _Get_current() const& noexcept {
+        return _Current;
+    }
+    constexpr iterator_type&& _Get_current() && noexcept {
+        return static_cast<iterator_type&&>(_Current);
     }
 
 private:
     iterator_type _Current{};
 };
 
 template <class _Iter1, class _Iter2>
-_NODISCARD _CONSTEXPR17 bool operator==(const move_iterator<_Iter1>& _Left, const move_iterator<_Iter2>& _Right)
+_NODISCARD _CONSTEXPR17 bool operator==(const move_iterator<_Iter1>& _Left,
+    const move_iterator<_Iter2>& _Right) noexcept(noexcept(_Left.base() == _Right.base())) /* strengthened */

```

Comment:
`_Implicitly_conver_to<bool>`? (Also on 3522.)

Reply:
It's unclear where 3522 is...

Reply:
It's clear to me where 3522 is, but I meant the _noexcept-specifier_ for the following operator which is on line 3552 and also is lacking an `_Implicitly_convert_to<bool>`.

Reply:
Pre-existing: all of these `move_iterator` comparisons that use `!` are broken. We need to implicitly convert to `bool` (which the constraints allow) and then apply `!`. I'll let you decide if you want to deal with this in this PR or file an issue to deal with it in a followup.

Reply:
Nope, calm down, Casey - your mouth is moving faster than your brain. We only apply `!` to **the `bool` result from another `move_iterator` comparison**, not to the result of any base iterator comparison.